pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -97,7 +97,10 @@ def reset(self) -> None:
         # track epoch output
         self._epoch_output = [[] for _ in range(self.batch_loop.num_active_optimizers(self.total_batch_idx))]
 
-        if not self.restarting:
+        failed = self.batch_progress.current.ready != self.batch_progress.current.completed
```

Comment:
Won't this be affected by `reset_on_restart` applied while restarting there: https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/loops/base.py#L218

Reply:
I didn't know this exists, why is it there in the first place? if progress magically resets while we are loading the state dict then we are not actually loading what we intend to.

Reply:
This was put in place to make sure progress is properly reset for current.

Reply:
But we don't want to restart in case of fault-tolerant training, unless we are actually past the last batch in the epoch. This information is not available at the time of loading the loop state dict

Reply:
removed the redundant ready != completed check